src/foundry/client/apps/forms/journal-page-sheet.d.ts

```diff
@@ -0,0 +1,292 @@
+import type { EditorView } from "prosemirror-view";
+import type Showdown from "showdown";
+import type { Editor } from "tinymce";
+import type { JOURNAL_ENTRY_PAGE_FORMATS } from "../../../common/constants.mjs.js";
+import type { CONST } from "../../../common/module.mjs.js";
+
+declare global {
+  /**
+   * The Application responsible for displaying and editing a single {@link JournalEntryPage} document.
+   *
+   * @typeParam TOptions - The type of the options object
+   */
+  class JournalPageSheet<TOptions extends JournalPageSheet.Options = JournalPageSheet.Options> extends DocumentSheet<
+    TOptions,
+    JournalEntryPage
+  > {
+    /**
+     * @param object - The JournalEntryPage instance which is being edited.
+     * @param options - Application options
+     */
+    constructor(object: JournalEntryPage, options?: Partial<TOptions>);
+
+    /**
+     * {@inheritdoc}
+     * @defaultValue
+     * ```typescript
+     * foundry.utils.mergeObject(super.defaultOptions, {
+     *   classes: ["sheet", "journal-sheet", "journal-entry-page"],
+     *   width: 600,
+     *   height: 680,
+     *   resizable: true,
+     *   closeOnSubmit: false,
+     *   submitOnClose: true,
+     *   viewPermission: CONST.DOCUMENT_OWNERSHIP_LEVELS.OBSERVER
+     * });
+     * ```
+     */
+    static get defaultOptions(): JournalPageSheet.Options;
+    get template(): string;
+    get title(): string;
+    // FIXME: this should be async, but the parent classes would need to be updated first
```

Comment:
We don't really use async in this repo because all our definitions are "ambient", in .d.ts files. If you try to you get this error:
```
'async' modifier cannot be used in an ambient context.ts(1040)
```

The rationale basically boils down to the fact that `foo(): Promise<Bar>` is not in any way discernible from `async foo(): Bar` (which should actually be `async foo(): Promise<Bar>` by our linting rules) and they felt `async` would mislead people into thinking you couldn't extend the class with a sync function that happens to return a promise of the right type.

I personally think it's silly but as we actually can't use async in an ambient context there's no need for these comments. Please remove them all.